.circleci/config.yml

```diff
@@ -99,38 +88,58 @@ jobs:
       - store_artifacts:
           path: test-results/hypertrace-ui
           destination: test-results
+  merge-publish:
+    executor: publisher
+    steps:
+      - setup_for_publishing
+      - run: ./gradlew dockerPushImages
+  release-publish:
+    executor: publisher
+    steps:
+      - setup_for_publishing
+      - run: ./gradlew publish dockerPushImages
   validate-charts:
     executor: helm
     steps:
       - checkout
       - run:
           name: Helm Charts Lint and Template Render
           command: |
+            helm dependency update ./helm/
             helm lint --strict ./helm/
             helm template ./helm/
-  publish:
-    executor: publisher
-    steps:
-      - setup_for_publishing
-      - run: ./gradlew :tag -Prelease
-      - run:
-          name: Push git tag
-          command: git push origin $(./gradlew -q :printVersion)
-      - run: ./gradlew dockerPushImages
-  package-charts:
+  release-charts:
     executor: helm
     steps:
       - checkout
+      - run:
+          name: Add release tag
+          command: |
+            git config --global user.email "${CIRCLE_USERNAME}@hypertrace.org"
+            git config --global user.name "$CIRCLE_USERNAME"
+            git tag -am "Released by $CIRCLE_USERNAME" $(git describe --abbrev=0 --tags | sed 's/^release-//')
+      - run:
+          name: Remove trigger tag
+          command: git tag -d release-$(git describe --abbrev=0)
       - run:
           name: Package and Publish Helm Charts
+          # Read the "name:" from Chart.yaml. The chart version is <chart-name>-<semver git tag>
           command: |
             CHART_VERSION=$(git describe --abbrev=0)
             CHART_NAME=$(awk '/^name:/ {print $2}' ./helm/Chart.yaml)
             export GOOGLE_APPLICATION_CREDENTIALS=${HOME}/helm-gcs-key.json
             echo ${HELM_GCS_CREDENTIALS} > ${GOOGLE_APPLICATION_CREDENTIALS}
+            helm dependency update ./helm/
             helm repo add helm-gcs ${HELM_GCS_REPOSITORY}
             helm package --version ${CHART_VERSION} --app-version ${CHART_VERSION} ./helm/
             helm gcs push ${CHART_NAME}-${CHART_VERSION}.tgz helm-gcs --public --retry
+      - add_ssh_keys:
+          fingerprints:
+            # This ssh key gives write permission needed for the following step.
+            - 'e3:b6:18:f1:68:bd:71:fe:9a:8b:08:aa:94:f5:1a:31'
```

Comment:
Wrong fingerprint, please remove the one for setup for publishing and sap it in here